nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'You must implement the correct_name method'
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    return @nameable.correct_name if @nameable.correct_name.length <= 10
+
+    @nameable.correct_name[0, 10]
```

Comment:
**[OPTIONAL]** Here you can use the [Ternary operator](https://syntaxdb.com/ref/ruby/ternary#:~:text=The%20ternary%20operator%20is%20used,behaving%20similar%20to%20a%20function.) it will simplify your code. Then it will be something like this: 

```
@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
```